source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/Util.mm

```diff
@@ -6,10 +6,104 @@
+        // bleed specification requires the object that's asked to be bled to have padding
+        if (collection->GetPadding()) {
+            std::shared_ptr<HostConfig> config = [acoConfig getHostConfig];
+            // bleed specification requires that there should be at leat one parental object with padding
+            if (collection->GetCanBleed()) {
+                InternalId internalId = collection->GetParentalId();
+                ACRContentStackView *view = (ACRContentStackView *)[rootView getBleedTarget:internalId];
+                // c++ to object-c enum conversion
+                if (view) {
+
+                    ACRBleedDirection direction = ACRRestricted;
+                    switch (collection->GetBleedDirection()) {
+                        case ContainerBleedDirection::BleedToLeading:
+                            direction = ACRToLeadingEdge;
+                            break;
+                        case ContainerBleedDirection::BleedToTrailing:
+                            direction = ACRToTrailingEdge;
+                            break;
+                        case ContainerBleedDirection::BleedToBothEdges:
+                            direction = ACRToBothEdges;
+                            break;
+                        default:
+                            direction = ACRRestricted;
+                            break;
+                    }
+                    // 1. create a background view (bv).
+                    // 2. bv is added to bleed target view (tv), which is also a parent view.
+                    // bv is then pinned to the tv according to the bleed direction
+                    // bv gets current container view's (cv) container style
+                    // and cv's container style is reset to transparent, such that
+                    // bv's container style will be diplayed.
+                    // container view's stack view (csv) holds content views, and bv dislpays container style
+                    // we transpose them, and get the final result
+                    UIView *backgroundView = [[UIView alloc] init];
+                    container.backgroundView = backgroundView;
+                    backgroundView.translatesAutoresizingMaskIntoConstraints = NO;                   
+                    [view addSubview:backgroundView];
+                    [view sendSubviewToBack:backgroundView];
+                    backgroundView.backgroundColor = container.backgroundColor;
+                    container.backgroundColor = UIColor.clearColor;
+                    [backgroundView.topAnchor constraintEqualToAnchor:container.topAnchor].active = YES;
+                    [backgroundView.bottomAnchor constraintEqualToAnchor:container.bottomAnchor].active = YES;
+
+                    // reset current width constraints and update them with new one according to bleed specification
+                    [container bleed:config->GetSpacing().paddingSpacing priority:1000 target:backgroundView
```

Comment:
>priority:1000  [](start = 73, length = 14)

do we ever `bleed` with a priority other than `1000`?

Reply:
1000 is the max value.

---
In reply to: [268272608](https://github.com/Microsoft/AdaptiveCards/pull/2551#discussion_r268272608) [](ancestors = 268272608)